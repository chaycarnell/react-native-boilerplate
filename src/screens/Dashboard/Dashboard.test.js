import { NavigationContext } from '@react-navigation/native';
import { cleanup, fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import DashboardScreen from './Dashboard';

afterEach(cleanup);

describe('Dashboard screen', () => {
  describe('given default props', () => {
    let renderedComponent;

    const testProps = {
      navigation: {
        navigate: jest.fn(),
      },
    };

    const navContext = {
      isFocused: () => true,
      addListener: jest.fn(() => jest.fn()),
    };

    beforeEach(() => {
      renderedComponent = render(
        <NavigationContext.Provider value={navContext}>
          <DashboardScreen navigation={testProps.navigation} />
        </NavigationContext.Provider>,
      );
    });

    it('should render correctly', () => {
      const { toJSON } = renderedComponent;
      expect(toJSON()).toMatchSnapshot();
    });

    it('should navigate to dashboard page when touchable highlight is selected', () => {
      const { getByTestId } = renderedComponent;
      fireEvent.press(getByTestId('DashboardNavigation'));

      expect(testProps.navigation.navigate).toHaveBeenCalled();
      expect(testProps.navigation.navigate).toHaveBeenCalledWith('Auth', {});
    });
  });
});
