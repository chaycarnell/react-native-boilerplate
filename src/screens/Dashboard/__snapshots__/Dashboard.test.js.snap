// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard screen given default props should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="DashboardNavigation"
  >
    <View
      color="#007d79"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#007d79",
            "display": "flex",
            "height": 667,
            "justifyContent": "center",
            "opacity": 1,
            "width": 750,
          },
        ]
      }
    >
      <Text
        color="#FFF"
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 48,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
      }
    }
  >
    Built for 
  </Text>
</View>
`;
